src/storage/kv_storage/littlefs/kv_littlefs.cpp

```diff
@@ -0,0 +1,143 @@
+/**
+ * This software is distributed under the terms of the MIT License.
+ * Copyright (c) 2023 LXRobotics.
+ * Author: Alexander Entinger <alexander.entinger@lxrobotics.com>
+ * Contributors: https://github.com/107-systems/107-Arduino-Cyphal-Support/graphs/contributors.
+ */
+
+/**************************************************************************************
+ * INCLUDE
+ **************************************************************************************/
+
+#include "kv_littlefs.h"
+
+#if __GNUC__ >= 11
+
+#include <map>
+#include <string>
+#include <variant>
+#include <sstream>
+
+/**************************************************************************************
+ * NAMESPACE
+ **************************************************************************************/
+
+namespace cyphal::support::platform::storage
+{
+
+/**************************************************************************************
+ * CONSTANTS
+ **************************************************************************************/
+
+static std::map<littlefs::Error, Error> const LITTLEFS_TO_STORAGE_ERROR_MAP =
+{
+  {littlefs::Error::IO          , Error::IO},
+  {littlefs::Error::CORRUPT     , Error::Internal},
+  {littlefs::Error::NOENT       , Error::Existence},
+  {littlefs::Error::EXIST       , Error::Existence},
+  {littlefs::Error::NOTDIR      , Error::Internal},
+  {littlefs::Error::ISDIR       , Error::API},
+  {littlefs::Error::NOTEMPTY    , Error::API},
```

Comment:
This error is returned when you attempt to remove a directory that is not empty, which also makes it an existence error.

Reply:
Fixed in fa3b6bfbb0ca85d89e97a0e72c19adb569ec4ab3 :+1: Thank you :bow: